docs/api/en/core/Object3D.html

```diff
@@ -314,6 +314,11 @@ <h3>[method:this remove]( [param:Object3D object], ... )</h3>
 		Removes *object* as child of this object. An arbitrary number of objects may be removed.
 		</p>
 
+		<h3>[method:this removeAll]()</h3>
+		<p>
+		Removes all child objects.
```

Comment:
`removeAll()`. Remove all what?

Does it remove the object? The object's children? The children of the children?

And how are they disposed?

This perhaps should be an app-level method, not a library method.

Reply:
Removes all children, but doesn't traverse the hierarchy.
I've found myself needed this often, but yes... the dispose situation can be confusing...

Reply:
I do not think it belongs in the library. The proper implementation is app-specific.

But at a minimum, I think the name should be changed to something else.

Reply:
Then we should probably change `remove()` to `removeChild()`, too. Or event to `removeChildButNotChildrenOfChild()`...

THB, I think the name is totally fine as long as the behavior is properly documented. This is something that could be improved to provide more clarity.

Reply:
> But at a minimum, I think the name should be changed to something else.

Neither Mugen87 nor I are native english speakers. Any help with naming is always welcome 🙏

Reply:
If the docs above are accurate, then perhaps `.removeChildren()` or `.removeChildObjects()`?

Reply:
I'm going to go with `.clear()` which is what I used in [ui.js](https://github.com/mrdoob/three.js/blob/dev/editor/js/libs/ui.js#L9-L61).

Reply:
https://github.com/mrdoob/three.js/commit/315d0fc94b054ef8fc98497d6c71e61b9c9a3715

Reply:
Interesting, this conflicts with `CubeCamera`:
https://github.com/mrdoob/three.js/blob/dev/src/cameras/CubeCamera.js#L97-L111

Reply:
Um this also breaks the TS build.

> src/cameras/CubeCamera.d.ts(16,2): error TS2416: Property 'clear' in type 'CubeCamera' is not assignable to the same property in base type 'Object3D'.

Reply:
Yep, I'll think about this.